test/Datadog.Trace.Tests/DiagnosticListeners/AspNetCoreDiagnosticObserverTests.cs

```diff
@@ -1,19 +1,46 @@
 #if !NETFRAMEWORK
 using System;
 using System.Collections.Generic;
+using System.Reflection;
+using System.Threading.Tasks;
 using Datadog.Trace.Agent;
 using Datadog.Trace.Configuration;
 using Datadog.Trace.DiagnosticListeners;
 using Datadog.Trace.Sampling;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
 using Microsoft.AspNetCore.Hosting.Internal;
 using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
 using Moq;
 using Xunit;
+using Xunit.Sdk;
 
 namespace Datadog.Trace.Tests.DiagnosticListeners
 {
+    [CollectionDefinition(nameof(AspNetCoreDiagnosticObserverTests), DisableParallelization = true)]
+    [TracerRestorer]
     public class AspNetCoreDiagnosticObserverTests
     {
+        [Fact]
+        public async Task<string> CompleteDiagnosticObserverTest()
+        {
+            Tracer.Instance = GetTracer();
+
+            var builder = new WebHostBuilder()
+                .UseStartup<Startup>();
+
+            var testServer = new TestServer(builder);
+            var client = testServer.CreateClient();
+
+            Datadog.Trace.ClrProfiler.Instrumentation.Initialize();
```

Comment:
`Datadog.Trace.ClrProfiler.Instrumentation.Initialize();`

I feel like it could become a problem if multiple tests rely on this. `DiagnosticManager.Instance` has an internal setter, can you use it to manage the DiagnosticsManager yourself and dispose it at the end of the test?

Reply:
fixed in https://github.com/DataDog/dd-trace-dotnet/pull/1061/commits/b948ceba50feaa8382129fe7dc7a06a1b5034835